packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -204,11 +207,16 @@ export class TerminalWidget extends BaseWidget implements StatefulWidget {
      * new terminal widget.
      * If id is provided attach to the terminal for this id.
      */
-    async start(id?: number): Promise<void> {
+    async start(id?: number): Promise<number> {
         this.terminalId = typeof id !== 'number' ? await this.createTerminal() : await this.attachTerminal(id);
         this.resizeTerminalProcess();
         this.connectTerminalProcess();
+        if (typeof this.terminalId === "number" && IBaseTerminalServer.validateId(this.terminalId)) {
```

Comment:
`typeof this.terminalId === "number"` is redundant in this case, `IBaseTerminalServer.validateId(this.terminalId)` already does it

Reply:
Yes code can be simplified, but terminalId has type "number | undefined", that why I applied redundant checking. If you don't mind, I changed type to simple "number".

Reply:
ok, you could also try to change `IBaseTerminalServer.validateId` that it narrows a type to `number`:
```ts
export namespace IBaseTerminalServer {
    export function validateId(id?: number): id is number {
        ...
    }
}
```